src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,110 @@
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            if (statement.executeUpdate() < 1) {
+                throw new RuntimeException("Was inserted 0 row, not at least one row");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        return Optional.ofNullable(get(id));
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                bookList.add(new Book(resultSet.getObject("id", Long.class),
+                        resultSet.getString("title"),
+                        resultSet.getBigDecimal("price")));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books", e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        int updatedRow;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            updatedRow = statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book's information for id = "
+                    + book.getId(), e);
+        }
+        return (updatedRow > 0) ? book : null;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        int deletedRow;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            deletedRow = statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete a book by id = " + id, e);
+        }
+        return deletedRow > 0;
+    }
+
+    private Book get(Long id) {
```

Comment:
why do we need this method?

Reply:
just to use this result for optional type - it looked concise: without creating a variable of Book. But I understand what you mean - I'll fix it.